### DISCUSSION TITLE
Long term statistics

### DISCUSSION THREAD

**Discussion by desosav** [DurgNomis-drol/ha_toyota#107] [Q&A] [answered]
Hello everyone,
i was wondering if it is possible to keep long term statistics for some of the values/sensors.
Let's take for example the daily/monthly/yearly average fuel consumption, where it is possible to see the avg value for the current month but not for the previous months. I can see the yearly consumption for 2022 but not for 2021. It would be great if there was a way to display that kind of info in a graph or a dashboard (pretty much how the energy dashboard works in HA)
By the way many thanks to the dev/maintainer of this integration. I really appreciate the effort and being able to incorporate car data and sensors in HA.

**Comment by DurgNomis-drol** [comment, marked as answer]
Sorry for the late answer, i have been moving and haven't found the time.

Please see [this](https://github.com/DurgNomis-drol/ha_toyota/discussions/30) discussion for inspiration on how to compare data.

Some sensor already have Home Assistant Long term statistics enabled.

**Reply by desosav** [reply]
Thanks for sharing. I totally missed that discussion topic.
LTSS and timescaledb should do the trick for long term storage. 
I only wish someone could share more sql queries/examples since i am not good at data querying.
But hey, as you said it is great for inspiration.

**Reply by DurgNomis-drol** [reply]
Just ask in that discussion, i know he is more then willing to share his knowledge.

Or search the HA forum for people using timescale and grafana, a lot of people wants to do the same as you 😊

I personally hope that HA at some point is gonna have the ability to compare data from history, that would make a lot of sense for an integration like this.
